### PR TITLE
delete APM_API_CONSUMER_APPS records as well when deleting a App

### PR DESCRIPTION
**Pull request by sajithab**
jira - https://wso2.org/jira/browse/APPM-791

### CODE REVIEW

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 1]
```diff
@@ -62,26 +27,22 @@
...
 import org.wso2.carbon.appmgt.impl.entitlement.EntitlementServiceFactory;
 import org.wso2.carbon.appmgt.impl.internal.ServiceReferenceHolder;
 import org.wso2.carbon.appmgt.impl.token.JWTGenerator;
-import org.wso2.carbon.appmgt.impl.utils.APIMgtDBUtil;
-import org.wso2.carbon.appmgt.impl.utils.AppManagerUtil;
-import org.wso2.carbon.appmgt.impl.utils.APIVersionComparator;
-import org.wso2.carbon.appmgt.impl.utils.LRUCache;
-import org.wso2.carbon.appmgt.impl.utils.RemoteUserManagerClient;
-import org.wso2.carbon.appmgt.impl.utils.URLMapping;
+import org.wso2.carbon.appmgt.impl.utils.*;
```
Comment (darshanasbg) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:45]: Avoid using wildcard imports when possible.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 2]
```diff
@@ -4679,6 +4655,10 @@ public void deleteAPI(APIIdentifier apiId) throws AppManagementException {
 			prepStmt.setInt(1, id);
 			prepStmt.execute();
 
+			prepStmt = connection.prepareStatement(deleteConsumerQuery);
+			prepStmt.setString(1, apiId.getApiName() + "-" + apiId.getVersion());
+			prepStmt.execute();
```
Comment (darshanasbg) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4660]: Need to close prepare statement in all places.

Comment (sajithab) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4660]: Yes, there is an issue not closing the statements properly. The issue was there in the original code, but I overlooked it. I will fix this in all relevant locations in this file.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 1 above)
Comment (sajithab) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:45]: Already fixed this with https://github.com/sajithab/carbon-appmgt/commit/a4d8c93f9fbe7595ea6ccb89fe60036ff9751055

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 2 above)
Comment (sajithab) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4660]: Fixed with https://github.com/wso2/carbon-appmgt/pull/150

**Pull request merged**